### ISSUE TITLE
[SR-15540] Documentation coverage "USR" column is named incorrectly

### ISSUE THREAD

**Issue by franklinsch** [swiftlang/swift-docc#205]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-15540      |
   |Radar             | None         |
   |Original Reporter | @franklinsch      |
   |Type              | Bug    |

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       | Swift-DocC    |
   |Labels            | Bug        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: 03bfd579fe69e2e3c621d898704baf08

  </details>

**Issue Description:**

When compiling documentation with the experimental documentation support enabled, the coverage data that's emitted includes a column called "USR". The naming is incorrect because the values that DocC emits is actually the documentation links of the generated pages, rather than their USR (the USR is a compiler-generated unique string that identifies a symbol). The column should be named to something like "Documentation Link" instead.

**Comment by sofiadinizms** [comment]
Hi, @franklinsch! I'm interested in working on this issue as my first contribution to the swift-docc repository and I was wondering if you or anyone else is currently working on it. It seems to be fairly simple, so even thought it's not tagged as a good-first-issue, I believe it can be accomplished.

**Comment by d-ronnqvist** [comment]
Hi @sofiadinizms. As far as I know no-one is working on this. I've assigned it to you to indicate that you're working on it.

**Comment by d-ronnqvist** [comment]
> The naming is incorrect because the values that DocC emits is actually the documentation links of the generated pages, rather than their USR (the USR is a compiler-generated unique string that identifies a symbol). The column should be named to something like "Documentation Link" instead.

If I can get pedantic, this value that's being reported here is the description of `ResolvedTopicReference` which is more of an "identifier" than a "link". To me, a "documentation link" would imply that developers can take that string to use it to reference a symbol in their documentation but that's not generally true for the string representation of a `ResolvedTopicReference`. 

IMO a name like "Page ID" would be more correct. We could also name it "Symbol ID" but there's a chance that some would misinterpret that and think that it holds USR values.

**Comment by sofiadinizms** [comment]
> IMO a name like "Page ID" would be more correct.

Yes, I agree. My initial ideas leaned more towards a name related to "link", however after reading your comment and checking the implementation of `ResolvedTopicReference`, it really does make more sense to use name it as some kind of "ID" (such as "Page ID", like you suggested)

**Comment by sofiaromorales** [comment]
What about `referencePath` or `referenceTopicPath`?  I don't have any strong opinion but AFAIK we don't use the term PageID in the codebase.

This is me just throwing ideas, don't feel constrained to name it like that 😄.

**Comment by sofiadinizms** [comment]
> What about `referencePath` or `referenceTopicPath`? I don't have any strong opinion but AFAIK we don't use the term PageID in the codebase.

Thank you for the suggestion! I think something along these lines would also make sense. Considering that the description of a ResolvedTopicReference contains the URL for the resolved topic reference's state and the requests for improved clarity in the titles of these columns, specially this USR one, I'm leaning towards naming it something like"topicPath" or "referenceTopicPath". 

In summary, I think all suggestions are fitting, but maybe highlighting the fact that it is an URL, while not leaving margin for associations with "linking", might make the meaning of the field more obvious. @d-ronnqvist I know that your suggestion diverged from this, but what do you think of this option?

Additionally, I think this issue will be done very soon and I would just like to confirm how to appropriately create a Pull Request associated with it. The [Contributing](https://github.com/swiftlang/swift-docc/blob/main/CONTRIBUTING.md#building-swift-docc) document briefly mentions the creation of a local branch, but there seems to be no mentions on branch naming and PR creation conventions. The "Create branch associated with this issue" GitHub button also doesn't seem to be available in this issue, so I would greatly appreciate if you had any documentation regarding this process (I searched for it in the project documentation and in the forums with no success, but maybe I missed something).

**Comment by d-ronnqvist** [comment]
> Additionally, I think this issue will be finished very soon, so I would just like to confirm how to appropriately create a Pull Request associated with it. The [Contributing](https://github.com/swiftlang/swift-docc/blob/main/CONTRIBUTING.md#building-swift-docc) document briefly mentions the creation of a local branch, but there seems to be no mentions of branch naming and PR creation conventions. 

You can name your branches however you want. The branch name is primarily for your own organization and won't matter for the pull request. 

> The "Create branch associated with this issue" GitHub button also doesn't seem to be available in this issue, so I would really appreciate if you had any documentation regarding this process. I searched for this in the project documentation and in the forums with no success, but maybe I missed something 🧐.

You can associate the pull request with the issue by referencing the issue number in the pull request description (`Resolved #12345`). Otherwise if you leave a link to the issue anywhere in the pull request someone with write permission can associate the issue and the pull request from the menus on the right hand side of the pull request UI.

Both approaches are described in the [Linking a pull request to an issue][1] GitHub documentation. 

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue

**Comment by d-ronnqvist** [comment]
Regarding the name I will say that this is a user facing string so it doesn't necessarily have to match the name of any type in the codebase. Its only needs to be clear and meaningful to the developer who reads the output data.
